src/StatusQ/Buttons/StatusTextWithIconButton.qml

```diff
@@ -0,0 +1,104 @@
+import QtQuick 2.14
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+
+Rectangle {
+    id: statusTextWithIconButton
+
+    property alias text: label.text
+    property alias font: label.font
+
+
+    property color iconNormalColor: "transparent"
+    property color iconDisabledColor: "transparent"
+
+    property color backgroundNormalColor: "transparent"
+    property color backgroundHoverColor: "transparent"
+    property color backgroundDisabledColor: "transparent"
+
+    property alias iconName: icon.icon
+
+    property bool showBackground: false
+
+    property bool loading: false
+
+    property real rightPadding: 18
+    property real leftPadding: 18
+
+    signal pressed(var mouse)
+
+
+    /// Implementation
+
+    radius: 8
+
+    color: {
+        if (enabled) {
+           return sensor.containsMouse ? backgroundHoverColor : backgroundNormalColor
+        } else {
+            return backgroundDisabledColor;
+        }
+    }
+
+    implicitWidth: layout.implicitWidth + rightPadding + leftPadding
+    implicitHeight: 38
+
+    MouseArea {
+        id: sensor
+
+        anchors.fill: parent
+        cursorShape: Qt.PointingHandCursor
+        hoverEnabled: !loading
+
+        Row {
+            id: layout
+
+            anchors.centerIn: parent
+
+            spacing: 4
+
+            StatusIcon {
+                id: icon
+                anchors.verticalCenter: parent.verticalCenter
+                visible: !loading
+                color: {
+                    if (loading) return "transparent"
+                    if (statusTextWithIconButton.enabled) {
+                        return iconNormalColor
+                    } else {
+                        return iconDisabledColor
+                    }
+                }
+            } // Icon
+
+            StatusBaseText {
+                id: label
+                anchors.verticalCenter: parent.verticalCenter
+                font.pixelSize: 13 // by design
+
+                color: {
+                    if (loading) return "transparent"
+                    if (statusTextWithIconButton.enabled) {
+                        return iconNormalColor
+                    } else {
+                        return iconDisabledColor
+                    }
+                }
+            } // Text
+        } // Row
+
+        StatusLoadingIndicator {
+            anchors.centerIn: parent
+            visible: loading
+            color: {
+                if (statusTextWithIconButton.enabled) {
+                    return iconNormalColor
+                } else {
+                    return iconDisabledColor
+                }
+            }
+        } // Indicator
+    } // Sennsor
+} // Rectangle
```

Comment:
This component should probably be just a composition of existing components.

I wonder if it would be feasible to implement a "slot" in `Status(Flat)Button` instead

Reply:
To elaborate, we could have something like:

```
StatusFlatButton {
  text: "some text"
  icon.name: "chat"
}
```

Reply:
I can do it, but i think its just one button in Figma.